### PR TITLE
Have identity resolver enabled by default

### LINKED ISSUE #57: Have IdentityResolver enabled by default
## Description

Identity resolver is a feature that was recently introduced where a DID can be resolved from the explorer dashboard.

## Motivation

The identity resolver is already working on the IOTA Mainnet, however on a private tangle, it needs to be enabled on the configuration file for the iota explorer. See https://github.com/iotaledger/explorer/blob/dev/client/DEPLOYMENT.md

We will have this feature enabled by default.

## Requirements

Write a list of what you want this feature to do.

1.  Have identityResolver enabled by default.

## Are you planning to do it yourself in a pull request?

Yes.

### PR DESCRIPTION
**Pull request by peterokwara**
# Description of change

The following pull request enables the IdentityResolver by default. The IdentityResolver helpes resolves DID's and closes #57 

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)

### 1. The first step was setting up a private-tangle up and running with a short Network ID. Instead of `private-tangle` as the network name, we have `tangle` as the network name. This involves changes in several places in the one click tangle.

- The config file of the autopeering https://github.com/iotaledger/one-click-tangle/blob/d27af3e93e167b5efb6b2b155d610ce9764b5d46/hornet-private-net/config/config-autopeering.json#L29 the coordinator https://github.com/iotaledger/one-click-tangle/blob/d27af3e93e167b5efb6b2b155d610ce9764b5d46/hornet-private-net/config/config-coo.json#L61 the node https://github.com/iotaledger/one-click-tangle/blob/d27af3e93e167b5efb6b2b155d610ce9764b5d46/hornet-private-net/config/config-node.json#L62 the spammer https://github.com/iotaledger/one-click-tangle/blob/d27af3e93e167b5efb6b2b155d610ce9764b5d46/hornet-private-net/config/config-spammer.json#L61 need to be changed to `tangle`
- When setting up the snapshot, the network name needs to be changed from `private-tangle` to `tangle` here:  https://github.com/iotaledger/one-click-tangle/blob/d27af3e93e167b5efb6b2b155d610ce9764b5d46/hornet-private-net/private-tangle.sh#L193
- Run the private-tangle by running `./private-tangle.sh install 30`

### 2. Once the private tangle is running, the next step is to have the explorer up and running. Some things needed to be configured.

- Enable the Identity resolver here https://github.com/iotaledger/one-click-tangle/blob/chrysalis/explorer/config/webapp.config.local.json by adding the line ` "identityResolverEnabled": true`
- For some reason, if this is set as localhost, the client is unable to reach the api https://github.com/iotaledger/one-click-tangle/blob/d27af3e93e167b5efb6b2b155d610ce9764b5d46/explorer/tangle-explorer.sh#L80 if you get this error, change the location of the endpoint from `localhost` to `your ip address` in the line of code.
- Have the explorer installed and running by running the following command `./tangle-explorer.sh install ../hornet-private-net/` it should copy the network configuration automatically from the already running private-tangle, running within the same server/machine.
- If you click on the tools section, Identity resolver should show up as one of the tools with it's own dashboard. 
image

### 3. To test if the identity resolver works, clone the identity.rs repository here https://github.com/iotaledger/identity.rs and we will modify one of the examples. Ensure you have rust installed. Go to the https://github.com/iotaledger/identity.rs/blob/dev/examples/account/create_did.rs file and modify the following lines

- For the config.rs code https://github.com/iotaledger/identity.rs/blob/dev/examples/account/config.rs change the network name, explorer url and the private node url
- Run the config.rs code by running `cargo run --example_config`
- You can click on the url generated or copy the did and check in the Identity resolver if it works

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by peterokwara** [comment]
There is a minor bug/warning that shows up that states **Selected network does not match DID network** when the network name is changed from **private-tangle** to **tangle**. Everything still works. It may have something to do with network configuration for the explorer.

**Comment by jmcanterafonseca-iota** [comment]
TL;DR

It is perfectly possible to use the Private Tangle one click together with IOTA Identity. For convenience purposes the name of the network under the explorer has been changed to `tangle`. Your IOTA DIDs should be of the form `iota:did_:tangle` and you will be able to browse your DIDs through the default-enabled identity resolver.
